### PR TITLE
modules doc updates

### PR DESCRIPTION
**Pull request by ebsmothers**
A bunch of miscellaneous changes to torchtune/modules docstrings so that our API docs look a bit nicer. Tbh I probably could have done a lot more here and many of the changes are more surface-level, but every little bit helps.

### CODE REVIEW

File: torchtune/modules/loss/ce_chunked_output_loss.py [hunk 1]
```diff
@@ -12,16 +12,16 @@
...
+    Whenever the model is trained with bf16, before running CE, we have to upcast
     it to fp32 for better accuracy and stability. When upcasting happens, the memory usage doubles.
     Models like llama3 have large vocabulary size and, therefore, have a large output
-    result (bsz, num_tokens, vocab_size). If we chunk on the token level, you can still compute
+    tensor of shape ``(bsz, num_tokens, vocab_size)``. If we chunk on the token level, you can still compute
     the cross entropy normally, but upcasting only one chunk at a time saves considerable memory.
 
     The CE and upcasting have to be compiled together for better performance.
-    When using this class, we recommend using torch.compile only on the method `compute_cross_entropy`.
+    When using this class, we recommend using :func:`torch.compile` only on the method ``compute_cross_entropy``.
```
Comment (RdoubleA) [inline torchtune/modules/loss/ce_chunked_output_loss.py:24]: Can you self reference method here?

File: torchtune/modules/lr_schedulers.py [hunk 2]
```diff
@@ -19,8 +19,8 @@ def get_cosine_schedule_with_warmup(
 ) -> LambdaLR:
     """
     Create a learning rate schedule that linearly increases the learning rate from
-    0.0 to lr over num_warmup_steps, then decreases to 0.0 on a cosine schedule over
-    the remaining num_training_steps-num_warmup_steps (assuming num_cycles = 0.5).
+    0.0 to lr over ``num_warmup_steps``, then decreases to 0.0 on a cosine schedule over
+    the remaining ``num_training_steps-num_warmup_steps`` (assuming num_cycles = 0.5).
```
Comment (RdoubleA) [inline torchtune/modules/lr_schedulers.py:23]: Backticks on num cycles

File: torchtune/modules/model_fusion/_fusion.py [hunk 3]
```diff
@@ -149,7 +149,7 @@ class FusionEmbedding(nn.Module):
     second embedding for the additional tokens. During forward this module routes
     the tokens to the appropriate embedding table.
 
-    Use this as a drop-in replacement for `nn.Embedding` in your model.
+    Use this as a drop-in replacement for ``nn.Embedding`` in your model.
```
Comment (RdoubleA) [inline torchtune/modules/model_fusion/_fusion.py:152]: Point to :class:torch.nn.Embedding?

File: torchtune/modules/position_embeddings.py [hunk 4]
```diff
@@ -85,16 +85,13 @@ def forward(
                 If none, assume the index of the token is its position id. Default is None.
 
         Returns:
-            torch.Tensor: output tensor with RoPE applied
+            torch.Tensor: output tensor with shape [b, s, n_h, h_d]
```
Comment (RdoubleA) [inline torchtune/modules/position_embeddings.py:88]: Backticks on shape

File: torchtune/modules/tied_linear.py [hunk 5]
```diff
@@ -31,4 +32,12 @@ def __init__(self, tied_module: nn.Module):
             )
 
     def __call__(self, x: torch.tensor) -> torch.tensor:
+        """
+        Args:
+            x (torch.tensor): Input tensor. Should have shape ``(..., in_dim)``, where ``in_dim``
```
Comment (RdoubleA) [inline torchtune/modules/tied_linear.py:37]: Why are these typed with lowercase tensor?

**Review by RdoubleA — APPROVED** [review]

File: torchtune/modules/loss/ce_chunked_output_loss.py (hunk 1 above)
Comment (ebsmothers) [reply torchtune/modules/loss/ce_chunked_output_loss.py:24]: Was debating it but honestly it's like 5 lines below and not well-documented anyways so I thought it was overkill

File: torchtune/modules/tied_linear.py (hunk 5 above)
Comment (ebsmothers) [reply torchtune/modules/tied_linear.py:37]: ugh you caught me. idk and idc, I just wanted these changes to be docstring-only

Comment (ebsmothers) [reply torchtune/modules/tied_linear.py:37]: Leaving it as is for now, we can check with @felipemello1 on whether this was by design or not

Comment (felipemello1) [reply torchtune/modules/tied_linear.py:37]: I did it just to see if you guys truly review the PRs, obviously. Congratulations, you passed and test! Wanna update it to upper case?

File: torchtune/modules/attention.py [hunk 6]
```diff
@@ -28,7 +28,7 @@ class MultiHeadAttention(nn.Module):
     Following is an example of MHA, GQA and MQA with num_heads = 4
 
     (credit for the documentation:
-    https://github.com/Lightning-AI/lit-gpt/blob/main/lit_gpt/config.py).
+    https://github.com/Lightning-AI/litgpt/blob/eda1aaaf391fd689664f95487ab03dc137e213fd/litgpt/config.py).
```
Comment (SalmanMohammadi) [inline torchtune/modules/attention.py:31]:
```suggestion
    `litgpt.Config <https://github.com/Lightning-AI/litgpt/blob/eda1aaaf391fd689664f95487ab03dc137e213fd/litgpt/config.py>`_).
```

File: torchtune/modules/layer_norm.py [hunk 7]
```diff
@@ -25,7 +25,7 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             x (torch.Tensor): Input tensor.
 
         Returns:
-            torch.Tensor: The normalized output tensor.
+            torch.Tensor: The normalized output tensor having the same shape as x.
```
Comment (SalmanMohammadi) [inline torchtune/modules/layer_norm.py:28]:
```suggestion
            torch.Tensor: The normalized output tensor having the same shape as ``x``.
```

File: torchtune/modules/model_fusion/_fusion.py [hunk 8]
```diff
@@ -356,8 +356,8 @@ def forward(
         KV values for each position.
 
         Returns:
-            Tensor: output tensor with shape [b x s x v] or a list of layer
-                output tensors defined by ``output_hidden_states`` with the
+            Tensor: output tensor with shape [b x s x v] or a list of layer \
```
Comment (SalmanMohammadi) [inline torchtune/modules/model_fusion/_fusion.py:359]:
```suggestion
            Tensor: Output tensor with shape ``[b x s x v]`` or a list of layer \
```

File: torchtune/modules/rlhf/rewards.py [hunk 9]
```diff
@@ -27,7 +27,7 @@ def get_reward_penalty_mask(
 
     Args:
         padding_masks (torch.Tensor): torch.Tensor where True indicates a padding token in the generated
-            sequence, and False otherwise. Shape: (b, reponse_len)
+            sequence, and False otherwise. Shape: (b, response_len)
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:30]:
```suggestion
            sequence, and False otherwise. Shape: ``(b, response_len)``
```

File: torchtune/modules/rlhf/rewards.py [hunk 10]
```diff
@@ -27,7 +27,7 @@ def get_reward_penalty_mask(
...
         seq_lens (torch.Tensor): The length of each generated sequence. Shape: (b,)
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:31]:
```suggestion
        seq_lens (torch.Tensor): The length of each generated sequence. Shape: ``(b,)``
```

File: torchtune/modules/rlhf/rewards.py [hunk 11]
```diff
@@ -58,8 +58,8 @@ def get_rewards_ppo(
 
     Args:
         scores (torch.Tensor): Reward model scores, shape (b,).
-        logprobs (torch.Tensor): Policy logprobs, shape (b, reponse_len).
-        ref_logprobs (torch.Tensor): Reference base model, shape (b, reponse_len).
+        logprobs (torch.Tensor): Policy logprobs, shape (b, response_len).
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:61]:
```suggestion
        logprobs (torch.Tensor): Policy logprobs, shape ``(b, response_len)``.
```

File: torchtune/modules/rlhf/rewards.py [hunk 12]
```diff
@@ -58,8 +58,8 @@ def get_rewards_ppo(
...
+        ref_logprobs (torch.Tensor): Reference base model, shape (b, response_len).
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:62]:
```suggestion
        ref_logprobs (torch.Tensor): Reference base model logprobs, shape ``(b, response_len)``.
```

File: torchtune/modules/rlhf/rewards.py [hunk 13]
```diff
@@ -188,19 +188,19 @@ def estimate_advantages(
     https://arxiv.org/pdf/1506.02438.pdf
 
     Args:
-        values (torch.Tensor): The predicted values for each state. Shape: (b, reponse_len)
-        rewards (torch.Tensor): The rewards received at each time step. Shape: (b, reponse_len)
+        values (torch.Tensor): The predicted values for each state. Shape: (b, response_len)
+        rewards (torch.Tensor): The rewards received at each time step. Shape: (b, response_len)
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:192]:
```suggestion
        rewards (torch.Tensor): The rewards received at each time step. Shape: ``(b, response_len)``
```

Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:192]:
```suggestion
        values (torch.Tensor): The predicted values for each state. Shape: ``(b, response_len)``
        rewards (torch.Tensor): The rewards received at each time step. Shape: ``(b, response_len)``
```

File: torchtune/modules/rlhf/rewards.py [hunk 14]
```diff
@@ -188,19 +188,19 @@ def estimate_advantages(
...
         gamma (float): The discount factor.
         lmbda (float): The GAE-Lambda parameter.
         masks (Optional[torch.Tensor]): A bool mask tensor, where True indicates the corresponding value in ``values``
             should participate in the mean calculation. Default None.
     Returns:
         Tuple[torch.Tensor, torch.Tensor]: A tuple containing the estimated advantages and returns.
-            - advantages (torch.Tensor): The estimated advantages. Shape: (b, reponse_len)
-            - returns (torch.Tensor): The estimated returns. Shape: (b, reponse_len)
+            - advantages (torch.Tensor): The estimated advantages. Shape: (b, response_len)
+            - returns (torch.Tensor): The estimated returns. Shape: (b, response_len)
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/rewards.py:200]:
```suggestion
            - advantages (torch.Tensor): The estimated advantages. Shape: ``(b, response_len)``
            - returns (torch.Tensor): The estimated returns. Shape: ``(b, response_len)``
```

File: torchtune/modules/rms_norm.py [hunk 15]
```diff
@@ -34,7 +34,7 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             x (torch.Tensor): input tensor to normalize
 
         Returns:
-            torch.Tensor: The output tensor after applying RMSNorm.
+            torch.Tensor: The normalized and scaled tensor having the same shape as x.
```
Comment (SalmanMohammadi) [inline torchtune/modules/rms_norm.py:37]:
```suggestion
            torch.Tensor: The normalized and scaled tensor having the same shape as ``x``.
```

File: torchtune/modules/tanh_gate.py [hunk 16]
```diff
@@ -22,6 +22,6 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             x (torch.Tensor): input tensor to gate
 
         Returns:
-            torch.Tensor: The output tensor after gating.
+            torch.Tensor: The output tensor after gating. Has the same shape as x
```
Comment (SalmanMohammadi) [inline torchtune/modules/tanh_gate.py:25]:
```suggestion
            torch.Tensor: The output tensor after gating. Has the same shape as ``x``.
```

maybe I'm getting carried away

File: torchtune/modules/tied_linear.py [hunk 17]
```diff
@@ -12,7 +12,8 @@
 class TiedLinear:
     """
     A tied linear layer, without bias, that shares the same weight as another linear layer.
-    This is useful for models that use tied weights, such as qwen and gemma.
+    This is useful for models that use tied weights, such as :func:`~torchtune.models.qwen2_0_5b`,
+    :func:`~torchtune.models.qwen2_1_5b` and :func:`~torchtune.models.gemma`.
```
Comment (SalmanMohammadi) [inline torchtune/modules/tied_linear.py:16]:
```suggestion
    :func:`~torchtune.models.qwen2_1_5b` and all of the :func:`~torchtune.models.gemma` models.
```

**Comment by SalmanMohammadi** [comment]
That was kind of cathartic
